main.go

```diff
@@ -168,13 +168,30 @@ func main() {
 		setupLog.Error(err, "Unable to get OperatorCondition")
 		os.Exit(1)
 	}
+	// apiclient.New() returns a client without cache.
+	// cache is not initialized before mgr.Start()
+	// we need this because we need to interact with OperatorCondition
+	apiClient, err := apiclient.New(mgr.GetConfig(), apiclient.Options{
+		Scheme: mgr.GetScheme(),
+	})
+	if err != nil {
+		setupLog.Error(err, "Unable to get Client")
+		os.Exit(1)
+	}
+
+	availCrds, err := getAvailableCRDNames(context.Background(), apiClient)
+	if err != nil {
+		setupLog.Error(err, "Unable to fetch CRD List")
```

Comment:
At this scope/context, you do not know how `getAvailableCRDNames` is implemented so the error message should not assume a list operation.
```suggestion
		setupLog.Error(err, "Unable get a list of available CRD names")
```